include/xtensor/xadapt.hpp

```diff
@@ -89,7 +89,7 @@ namespace xt
      */
     template <class C, std::size_t N, layout_type L = DEFAULT_LAYOUT>
     xtensor_adaptor<C, N, L>
-    xadapt(C& container, const std::array<typename C::size_type, N>& shape, layout_type l = L);
+    xadapt(C& container, const stat_shape<typename C::size_type, N>& shape, layout_type l = L);
```

Comment:
Sorry, I just discovered your new overload further down. Unfortunately it doesn't seem to be picked up. I'd love to investigate more, but lack the time (it's on Ubuntu 16.04 / GCC 5.4).